packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -48,13 +48,37 @@ class CupertinoPicker extends StatefulWidget {
     this.scrollController,
     @required this.itemExtent,
     @required this.onSelectedItemChanged,
-    @required this.children,
-  }) : assert(diameterRatio != null),
-       assert(diameterRatio > 0.0, RenderListWheelViewport.diameterRatioZeroMessage),
-       assert(magnification > 0),
-       assert(itemExtent != null),
-       assert(itemExtent > 0),
-       super(key: key);
+    @required List<Widget> children,
+  }) :  assert(children != null),
+        assert(diameterRatio != null),
+        assert(diameterRatio > 0.0, RenderListWheelViewport.diameterRatioZeroMessage),
+        assert(magnification > 0),
+        assert(itemExtent != null),
+        assert(itemExtent > 0),
+        childDelegate = new SliverChildListDelegate(children, addRepaintBoundaries: false, addAutomaticKeepAlives: false),
+        super(key: key);
+
+  /// Create picker from a lazy builder.
```

Comment:
This needs more doc. Which parameters are mandatory, refer to performance implications if you use the other constructor vs this one.  Document defaults. Each document should stand on its own. Do not assume users will read the document for the class and the other constructor before reading this one.

Reply:
Sorry this is lame ._. I document pretty poorly and mostly focus on getting the logic done because I'm not sure if I'm doing it optimistically. You can just ignore documenting because most likely I will come back for it later.